### ISSUE TITLE
[BUG]etcd config source not working properly in case key and its prefix both exist

### ISSUE THREAD

**Issue by seaguest** [go-micro/go-micro#2386]
**Describe the bug**

1. What are you trying to do?
when I have a key and its prefix as another key, the etcd config source is not working properly.

2. What did you expect to happen?
```

etcdctl get /conf/a
/conf/a
{
  "k": "a"
}

etcdctl get /conf/a/a1
/conf/a/a1
{
  "k": "a1"
}

I have two keys as above, I tried to get its data by their keys, I expect each key return its own value.
```

3. What happens instead?
I have two keys as above, I tried to get its data by their keys, but it always return 
{"k":"a1"}

**How to reproduce the bug:**

```
 package main

import (
	"flag"
	"github.com/asim/go-micro/plugins/config/source/etcd/v3"
	conf "github.com/asim/go-micro/v3/config"
	"github.com/seaguest/log"
)

func main() {
	regAddr := flag.String("registry", "127.0.0.1:2379", "etcd registry")
	key := flag.String("key", "/key", "etp storage config key")
	flag.Parse()

	etcdSource := etcd.NewSource(
		etcd.WithAddress(*regAddr),
		etcd.WithPrefix(*key),
		etcd.StripPrefix(true),
	)

	log.Error("start read config...")
	err := conf.Load(etcdSource)
	if err != nil {
		log.Error(err)
		return
	}

	log.Error(string(conf.Bytes()))
}

```
**Environment:**
Go Version: please paste `go version` output here
```
GO111MODULE="on"
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/*/.cache/go-build"
GOENV="/home/*/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE="*.wuren.com"
GOMODCACHE="/opt/golang/pkg/mod"
GONOPROXY="*.wuren.com,*.sublive.app"
GONOSUMDB="*.wuren.com,*.sublive.app"
GOOS="linux"
GOPATH="/opt/golang"
GOPRIVATE="*.wuren.com,*.sublive.app"
GOPROXY="https://goproxy.cn,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.17"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/*/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build1466580043=/tmp/go-build -gno-record-gcc-switches"

```

**Comment by seaguest** [comment]
The reason is here:
```
func (c *etcd) Read() (*source.ChangeSet, error) {
	if c.cerr != nil {
		return nil, c.cerr
	}

	rsp, err := c.client.Get(context.Background(), c.prefix, clientv3.WithPrefix())
	if err != nil {
		return nil, err
	}

```

if I change
```
	rsp, err := c.client.Get(context.Background(), c.prefix, clientv3.WithPrefix())
to
	rsp, err := c.client.Get(context.Background(), c.prefix)

```
it works fine, so I don't get why we need WithPrefix, I think it should be removed.

**Comment by xpunch** [comment]
I think you're right, can you make a PR to [v3](https://github.com/asim/go-micro/tree/v3)?

**Comment by xpunch** [comment]
Fixed in v3.7.1
```
go get github.com/asim/go-micro/plugins/config/source/etcd/v3@v3.7.1
```
